### ISSUE TITLE
Add support for serializing fields to Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer

### ISSUE THREAD

**Issue by sliedig** [aws/aws-lambda-dotnet#696]
## Description

I have a very simple Lambda function that returns a simple class. When I use the `Amazon.Lambda.Serialization.Json.JsonSerializer` decorator on my function (Version="1.5.0") I get the expected result. Switching to the new `Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer` (Version="2.0.1") I get an empty object returned.

The same behaviour is exhibited using sam local (using SAM CLI, version 1.0.0rc1).

Attached source:  [HelloWorld.zip](https://github.com/aws/aws-lambda-dotnet/files/4890404/HelloWorld.zip)

### Logs

No logs

### Environment

* Build Version: 2.0.1
* OS Info: OSX Mojave
* Build Environment: VSCode + Terminal `dotnet`
* Targeted .NET Platform:  dotnet core 3.1.300

## Resolution
- [ ] :wave: I can/would-like-to implement a fix for this problem myself
[HelloWorld.zip](https://github.com/aws/aws-lambda-dotnet/files/4890404/HelloWorld.zip)

---

This is a :bug: bug-report

**Comment by normj** [comment]
System.Text.Json which Amazon.Lambda.Serialization.SystemTextJson depends on doesn't support serializing fields. 

https://github.com/dotnet/runtime/issues/876

Till it does this library won't be able to support fields.

**Comment by NGL321** [comment]
@sliedig,

I have updated the issue title to reflect that it is a feature-request (to better guide resolution). Since this is blocked by an issue in a related 3rd party library, we will have to wait until their fix is released. Once that is done, feel free to submit a Pull Request here, OR someone from the team will take a look at it when able.

😸 😷 

@normj it appears that a [fix was implemented](https://github.com/dotnet/runtime/pull/36986) on the dotnet/runtime library two days ago, so this should be included on the next release.
